README.md

```diff
@@ -1,113 +1,217 @@
+| GEOSERVER_DATASTORE | string | GeoServer datastore name | `"polygonParts"` |
+
+### Ingestion Configuration
+
+| Variable Name | Type | Description | Default |
+|--------------|------|-------------|---------|
+| INGESTION_POLLING_INIT_TASK | string | Init task type for polling | `"init"` |
+| INGESTION_POLLING_FINALIZE_TASK | string | Finalize task type for polling | `"finalize"` |
+| INGESTION_NEW_JOB_TYPE | string | New ingestion job type | `"Ingestion_New"` |
+| INGESTION_UPDATE_JOB_TYPE | string | Update ingestion job type | `"Ingestion_Update"` |
+| INGESTION_SWAP_UPDATE_JOB_TYPE | string | Swap update job type | `"Ingestion_Swap_Update"` |
+| INGESTION_SEED_JOB_TYPE | string | Seed job type | `"Ingestion_Seed"` |
+| TILES_MERGING_TASK_TYPE | string | Tiles merging task type | `"tiles-merging"` |
+| TILES_MERGING_TILE_BATCH_SIZE | number | Batch size for tile merging | `10000` |
+| TILES_MERGING_TASK_BATCH_SIZE | number | Batch size for task merging | `5` |
+| TILES_SEEDING_TASK_TYPE | string | Tiles seeding task type | `"tiles-seeding"` |
+| TILES_SEEDING_GRID | string | Grid configuration for tiles seeding | `"WorldCRS84"` |
+| TILES_SEEDING_MAX_ZOOM | string | Maximum zoom level for seeding | `21` |
+| TILES_SEEDING_SKIP_UNCACHED | boolean | Skip uncached tiles during seeding | `true` |
+
+
+# Core Functionality
+
+- **Task Processing:**
+  - Handles two primary task types: "init" and "finalize"
+  - Supports multiple ingestion scenarios: new ingestion, update, and swap-update
+  - Implements task resumption capabilities for failure recovery
+
+- **Init Task:**
+  - Creates merge tasks for different ingestion job types
+  - Processes job parameters and polygon part metadata
+  - Manages task status transitions
+  - Creates merge tasks for ingestion job types using @map-colonies/mc-utils
+
+- **Finalize Task:**
+  - For New Ingestion:
+    - Handles layer naming convention:
+    	-nativeName:{productId}_{productType} for geoserver use
+    	-name:{productId}-{productType} for mapproxy use
+    - Manages layer insertion across:
+    	1. Inserts layer to MapProxy
+  	2. Inserts layer to GeoServer
+  	3. Inserts layer to Catalog
+    - Processes aggregated part data
+    - Manages job completion
+  - For Update Ingestion:
+    - Updates catalog layers
+    - Creates seed jobs and tasks
+  - For Swap Update Ingestion:
+    - Updates catalog layers
+    - Handles layer updates in MapProxy when required
+    - Creates seed jobs and tasks
+
+### Job Types Support
+
+- Ingestion_New
+- Ingestion_Update
+- Ingestion_Swap_Update
+
+Each job type supports both init and finalize task phases, with specialized handling for tiles merging operations through the tiles-merger consumer.
+
+## Architecture
+
+The service is part of a larger system that includes:
+- Job Manager
+- Ingestion Trigger
+- MapProxy API
+- GeoServer API
+- Raster Catalog Manager
+- Polygon Parts Manager
+- Tiles Merger
+
+## Dependencies
+
+Key dependencies include:
```

Comment:
Not sure we want versions here, otherwise we need update it every time versions are updated

Reply:
done